lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -194,8 +194,402 @@ local function boneVerify(self, bone)
+		end
+
+		if e2TypeID == TYPE_ANGLE or e2TypeID == TYPE_COLOR or e2TypeID == TYPE_VECTOR or e2TypeID == TYPE_MATRIX then return e2Value:ToTable() end
+
+		return nil
+	end,
+	[TYPE_ENTITY] = function(e2Value) -- TYPE_ENTITY from 'entity'
+		if TypeID(e2Value) == TYPE_ENTITY then return e2Value end
+
+		return nil
+	end,
+	[TYPE_VECTOR] = function(e2Value) -- TYPE_VECTOR from 'vector' or 'itable'
+		local e2TypeID = TypeID(e2Value)
+		if e2TypeID == TYPE_VECTOR then return e2Value end
+		if e2TypeID == TYPE_TABLE and isnumber(e2Value[1]) and isnumber(e2Value[2]) and isnumber(e2Value[3]) then return Vector(e2Value[1], e2Value[2], e2Value[3]) end
+
+		return nil
+	end,
+	[TYPE_ANGLE] = function(e2Value) -- TYPE_ANGLE from 'angle' or 'itable'
+		local e2TypeID = TypeID(e2Value)
+		if e2TypeID == TYPE_ANGLE then return e2Value
+		elseif e2TypeID == TYPE_TABLE and isnumber(e2Value[1]) and isnumber(e2Value[2]) and isnumber(e2Value[3]) then return Angle(e2Value[1], e2Value[2], e2Value[3]) end
+
+		return nil
+	end,
+	[TYPE_DAMAGEINFO] = function(e2Value) -- TYPE_DAMAGEINFO from 'damageinfo'
+		if TypeID(e2Value) == TYPE_DAMAGEINFO then return e2Value end
+	end,
+	[TYPE_EFFECTDATA] = function(e2Value) -- TYPE_EFFECTDATA from 'effectdata'
+		if TypeID(e2Value) == TYPE_EFFECTDATA then return e2Value end
+	end,
+	[TYPE_MATERIAL] = function(e2Value) -- TYPE_MATERIAL from 'string' or 'itable'
+		local e2TypeID = TypeID(e2Value)
+		if e2TypeID == TYPE_STRING then return Material(e2Value) end
+		if e2TypeID == TYPE_TABLE then -- Png parameters support
+			if #e2Value ~= 2 then return nil end
+
+			if TypeID(e2Value[1]) ~= TYPE_STRING then return nil end
+			if TypeID(e2Value[2]) ~= TYPE_STRING then return nil end
+
+			return Material(e2Value[1], e2Value[2])
+		end
+
+		return nil
+	end,
+	[TYPE_MATRIX] = function(e2Value) -- TYPE_MATRIX from 'matrix4'
+		if TypeID(e2Value) ~= TYPE_TABLE then return nil end
+
+		if #e2Value == 16 then
+			for i = 1, 16 do
+				if not isnumber(e2Value[i]) then return nil end
+			end
+
+			return Matrix({e2Value[1], e2Value[2], e2Value[3], e2Value[4]}, {e2Value[5], e2Value[6], e2Value[7], e2Value[8]}, {e2Value[9], e2Value[10], e2Value[11], e2Value[12]}, {e2Value[13], e2Value[14], e2Value[15], e2Value[16]})
+		end
+
+		return nil
+	end,
+	[TYPE_COLOR] = function(e2Value) -- TYPE_COLOR from 'vector' or 'vector4' or 'itable' or 'table'
+		local e2TypeID = TypeID(e2Value)
+		if e2TypeID == TYPE_VECTOR then
+			return Color(e2Value[1], e2Value[2], e2Value[3])
+		elseif e2TypeID == TYPE_TABLE then -- vector4 support + direct table support
+			if isnumber(e2Value[1]) and isnumber(e2Value[2]) and isnumber(e2Value[3]) and isnumber(e2Value[4]) then
+				return Color(e2Value[1], e2Value[2], e2Value[3], e2Value[4])
+			elseif e2Value.r and e2Value.g and e2Value.b then
+				if e2Value.a then return Color(e2Value.r, e2Value.g, e2Value.b, e2Value.a) end
+				return Color(e2Value.r, e2Value.g, e2Value.b)
+			end
+		end
+
+		return nil
+	end,
+}
```

Comment:
This should be e2 core code rather than local to prop lib.

Reply:
Yeah, I wanted to make it E2Lib code, but wasn't sure, and never got reply in Discord. Will move it